python/looker_sdk/rtl/requests_transport.py

```diff
@@ -64,20 +65,19 @@ def request(
         headers: Optional[MutableMapping[str, str]] = None,
         transport_options: Optional[transport.TransportSettings] = None,
     ) -> transport.Response:
-
-        url = f"{self.api_path}{path}"
+        request_path = self.make_path(path, authenticator)
```

Comment:
I don't think we need a separate function. https://refactoring.com/catalog/inlineFunction.html used to have more explanation but basically it's easier to just read

```suggestion
        url = path if urllib.parse.urlparse(path).scheme else f"{self.api_path}{path}"
```

inline then have to context switch to read the implementation of `make_path()`.

for sure we want to break up longer functions where it makes sense (and this function is definitely at my comfort edge of length) but because this logic only happens once and it's essentially one line it's better left inline.

Reply:
Just last night I was thinking that "make path" is the wrong function name and I intend to change it for the Typescript implementation. It's really "make url." The function takes the parameters, then provides the fully specified URL as output. That's the intent. It makes things like proxy sessions much easier to accomplish. The Typescript branch I've updated last night shows this.

Reply:
@jkaster I have a feeling you and I should talk about what the desired behavior is :-) I don't fully understand the motivations to make the transport layer directly usable by sdk callers, at least how it's implemented in python currently